tools/fv_restart.F90

```diff
@@ -167,25 +169,28 @@ subroutine fv_restart(fv_domain, Atm, dt_atmos, seconds, days, cold_start, grid_
           write(fname_ne,'(A, I2.2, A)') 'INPUT/fv_BC_ne.res.nest', Atm(n)%grid_number, '.nc'
           write(fname_sw,'(A, I2.2, A)') 'INPUT/fv_BC_sw.res.nest', Atm(n)%grid_number, '.nc'
           if (is_master()) print*, 'Searching for nested grid BC files ', trim(fname_ne), ' ', trim (fname_sw)
-          do_read_restart = file_exist(fname, Atm(n)%domain)
-          do_read_restart_bc = file_exist(fname_ne, Atm(n)%domain) .and. file_exist(fname_sw, Atm(n)%domain)
+          do_read_restart = file_exists(fname)
+          do_read_restart_bc = file_exists(fname_ne) .and. file_exists(fname_sw)
           if (is_master()) then
              print*, 'FV_RESTART: ', n, do_read_restart, do_read_restart_bc
              if (.not. do_read_restart_bc) write(*,*) 'BC files not found, re-generating nested grid boundary conditions'
           endif
           Atm(N)%neststruct%first_step = .not. do_read_restart_bc
        else
           fname='INPUT/fv_core.res.nc'
-          do_read_restart = file_exist('INPUT/fv_core.res.nc') .or. file_exist('INPUT/fv_core.res.tile1.nc')
+          do_read_restart = file_exists('INPUT/fv_core.res.nc') .or. file_exists('INPUT/fv_core.res.tile1.nc')
```

Comment:
I am not sure about this. 
If you are doing an ensemble run (i.e SPEAR), the restarts are named "INPUT/fv_core.ensXX.res.nc", but `file_exists` is only going to check for 'INPUT/fv_core.res.nc', so do_read_restart is going to be set to false...

Reply:
@uramirez8707 Would an fms2io open_file() call  adequately check for .ensXX files?

Reply:
yes I think that could work, with is_restart=.true.  and you can use the `FmsNetcdfFile_t` fileobj